shell/utils/cluster.js

```diff
@@ -44,3 +44,49 @@ export function filterHiddenLocalCluster(mgmtClusters, store) {
     return !target.isLocal;
   });
 }
+
+const clusterNameSegments = /([A-Za-z]+|\d+)/g;
+
+/**
+ * Shortens an input string based on the number of segments it contains.
+ * @param {string} input - The input string to be shortened.
+ * @returns {string} - The shortened string.
+ * @example smallIdentifier('local') => 'lcl'
+ * @example smallIdentifier('word-wide-web') => 'www'
+ */
+export function abbreviateClusterName(input) {
+  if (!input) {
+    return '';
+  }
+
+  if (input.length <= 3) {
+    return input;
+  }
+
+  const segments = input.match(clusterNameSegments);
+
+  if (!segments) return ''; // In case no valid segments are found
+
+  let result = '';
+
+  switch (segments.length) {
+  case 1: {
+    const word = segments[0];
+
+    result = `${ word[0] }${ word[Math.floor(word.length / 2)] }${ word[word.length - 1] }`;
+    break;
+  }
+  case 2: {
+    const w1 = `${ segments[0][0] }`;
+    const w2 = `${ segments[0].length >= 2 ? segments[0][segments[0].length - 1] : segments[1][0] }`;
+    const w3 = `${ segments[1][segments[1].length - 1] }`;
+
+    result = w1 + w2 + w3;
+    break;
+  }
+  default:
+    result = segments.slice(0, 3).map((segment) => segment[0]).join('');
```

Comment:
In a different PR i think we need to rework this. I added tests containing the following results

```
    ['customer-support-team-1', 'cst'],
    ['customer-support-team-one', 'cst'],
    ['customer-support-team-prod', 'cst'],
    ['customer-support-team-dev', 'cst'],
    ['customer-support-team-prod-1', 'cst'],
    ['customer-support-team-dev-1', 'cst'],
    ['customer-support-team-prod-2', 'cst'],
    ['customer-support-team-dev-2', 'cst'],
```

So all of above are the same. I think we need to consider, for the case of more than 2 segments something like...

<first segments first chart> <second segments first char> <last segments first char>

which would give

```
    ['customer-support-team-1', 'cs1'],
    ['customer-support-team-one', 'cso'],
    ['customer-support-team-prod', 'csp'],
    ['customer-support-team-dev', 'csd'],
    ['customer-support-team-prod-1', 'cs1'],
    ['customer-support-team-dev-1', 'cs1'],
    ['customer-support-team-prod-2', 'cs2'],
    ['customer-support-team-dev-2', 'cs2'],
```

so not perfect, but at least if the user creates a cluster with an important deliminator at the end we'll show it